src/main/java/org/testng/internal/annotations/AnnotationHelper.java

```diff
@@ -317,5 +319,41 @@ private static String createMethodKey(Method m) {
 
     return result.toString();
   }
+  
+  /**
+   * @return An array of all locally declared methods or equivalent thereof
+   * (such as default methods on Java 8 based interfaces that the given class
+   * implements).
+   */
+  private static Method[] getLocalMethods(Class<?> clazz) {
+    Method[] result;
+    Method[] declaredMethods = clazz.getDeclaredMethods();
+    List<Method> defaultMethods = getDefaultMethods(clazz);
+    if (!defaultMethods.isEmpty()) {
+      result = new Method[declaredMethods.length + defaultMethods.size()];
+      System.arraycopy(declaredMethods, 0, result, 0, declaredMethods.length);
+      int index = declaredMethods.length;
+      for (Method defaultMethod : defaultMethods) {
+        result[index] = defaultMethod;
+        index++;
+      }
+    }
+    else {
+      result = declaredMethods;
+    }
+    return result;
+  }
+
+  private static List<Method> getDefaultMethods(Class<?> clazz) {
+    List<Method> result = new LinkedList<Method>();
```

Comment:
I'm assuming these functions will have to be moved into a helper and used more widely so I kept the optimizations from Spring in (minus the change I made for juherr) and kept the return value congruent with `getDeclaredMethods`. Will be happy to change if you prefer readability.